app/containers/List/ListPicker.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { useState } from 'react';
+import { Text } from 'react-native';
+
+import { TActionSheetOptionsItem, useActionSheet } from '../ActionSheet';
+import { CustomIcon } from '../CustomIcon';
+import * as List from './index';
+import I18n from '../../i18n';
+import { useTheme } from '../../theme';
+import sharedStyles from '../../views/Styles';
+
+type TOption = { label: string; value: string; second?: string };
+type TOptions = { [key: string]: TOption[] };
+
+interface IListPicker {
+	title: string;
+	testID: string;
+	value?: string;
+	options: TOptions;
+	preference: string;
+	onChangeValue: (pref: string, param: { [key: string]: string }, onError: () => void) => void;
+}
+
+const ListPicker = ({ preference, options, value, title, testID, onChangeValue }: IListPicker) => {
+	const { showActionSheet, hideActionSheet } = useActionSheet();
+	const { colors } = useTheme();
+
+	const pref = value ? options[preference].find(option => option.value === value) : options[preference][0];
+
+	const [option, setOption] = useState(pref);
+
+	const actionSheetOptions: TActionSheetOptionsItem[] = options[preference].map(i => ({
+		title: I18n.t(i.label, { defaultValue: i.label, second: i.second }),
+		onPress: () => {
+			hideActionSheet();
+			onChangeValue(preference, { [preference]: i.value.toString() }, () => setOption(option));
+			setOption(i);
+		},
+		right: option?.value === i.value ? () => <CustomIcon name={'check'} size={20} color={colors.tintActive} /> : undefined
+	}));
+
+	return (
+		<List.Item
+			title={title}
+			testID={testID}
+			onPress={() => showActionSheet({ options: actionSheetOptions })}
+			right={() => (
+				<Text style={[{ ...sharedStyles.textRegular, fontSize: 16 }, { color: colors.actionTintColor }]}>
```

Comment:
`{ ...sharedStyles.textRegular, fontSize: 16 }` should have its own style rather than inline